tornado/test/httpserver_test.py

```diff
@@ -27,6 +27,25 @@
 from io import BytesIO
 
 
+def read_stream_start_line(stream, callback):
+    """Reads an HTTP response from `stream` and runs callback with its
+    headers and body."""
```

Comment:
"start line, headers, and body"

This is almost identical to read_stream_body. Can we consolidate them? We could implement read_stream_body in terms of read_stream_start_line, but there aren't that many existing uses of read_stream_body so I'd just make that save the start line too and update all the callers.